From 160dd61a0b2892e483e1383a7f14f010b9cea708 Mon Sep 17 00:00:00 2001
From: gurrev01 <gururaj.revankar@arm.com>
Date: Thu, 22 Jan 2026 06:37:23 +0000
Subject: [PATCH] usb_host_xhci_plat

Signed-off-by: gurrev01 <gururaj.revankar@arm.com>
---
 drivers/usb/host/xhci-plat.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/usb/host/xhci-plat.c b/drivers/usb/host/xhci-plat.c
index ecaa75718..39f8cca49 100644
--- a/drivers/usb/host/xhci-plat.c
+++ b/drivers/usb/host/xhci-plat.c
@@ -566,6 +566,7 @@ EXPORT_SYMBOL_GPL(xhci_plat_pm_ops);
 static const struct acpi_device_id usb_xhci_acpi_match[] = {
 	/* XHCI-compliant USB Controller */
 	{ "PNP0D10", },
+	{ "PNP0D15", },
 	{ }
 };
 MODULE_DEVICE_TABLE(acpi, usb_xhci_acpi_match);
-- 
2.43.0

