From d06c0865e5c5fcf4d28d9e2f220e7e985dabb3ad Mon Sep 17 00:00:00 2001
From: Amrathesh <amrathesh@arm.com>
Date: Wed, 22 Dec 2021 17:06:51 +0530
Subject: [PATCH] EDK2 BSA Patch

Signed-off-by: Amrathesh <amrathesh@arm.com>
---
 ShellPkg/ShellPkg.dsc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/ShellPkg/ShellPkg.dsc b/ShellPkg/ShellPkg.dsc
index a8b6de3..85dc2c0 100644
--- a/ShellPkg/ShellPkg.dsc
+++ b/ShellPkg/ShellPkg.dsc
@@ -60,6 +60,8 @@
   DxeServicesTableLib|MdePkg/Library/DxeServicesTableLib/DxeServicesTableLib.inf
   DxeServicesLib|MdePkg/Library/DxeServicesLib/DxeServicesLib.inf
   ReportStatusCodeLib|MdePkg/Library/BaseReportStatusCodeLibNull/BaseReportStatusCodeLibNull.inf
+  BsaValLib|ShellPkg/Application/bsa-acs/val/BsaValLib.inf
+  BsaPalLib|ShellPkg/Application/bsa-acs/pal/uefi_acpi/BsaPalLib.inf
 
 [LibraryClasses.ARM,LibraryClasses.AARCH64]
   #
@@ -98,6 +100,7 @@
   ShellPkg/Library/UefiShellDebug1CommandsLib/UefiShellDebug1CommandsLib.inf
   ShellPkg/Library/UefiShellNetwork1CommandsLib/UefiShellNetwork1CommandsLib.inf
   ShellPkg/Library/UefiShellNetwork2CommandsLib/UefiShellNetwork2CommandsLib.inf
+  ShellPkg/Application/bsa-acs/uefi_app/BsaAcs.inf
 
   ShellPkg/Application/Shell/Shell.inf {
     <PcdsFixedAtBuild>
-- 
2.17.1

