diff --git a/apps/uefi/sbsa_main.c b/apps/uefi/sbsa_main.c
index 9beb188..2af3e4d 100644
--- a/apps/uefi/sbsa_main.c
+++ b/apps/uefi/sbsa_main.c
@@ -168,6 +168,7 @@ execute_tests()
         goto exit_acs;
     }
 
+    val_print(INFO, "\n\n SystemReady band ACS v3.1.1", 0);
     val_print(INFO, "\n\n SBSA Architecture Compliance Suite\n");
     val_print(INFO, "    Version %d.", SBSA_ACS_MAJOR_VER);
     val_print(INFO, "%d.", SBSA_ACS_MINOR_VER);
