diff --git a/apps/uefi/sbsa_main.c b/apps/uefi/sbsa_main.c
index d684462..7f9c883 100644
--- a/apps/uefi/sbsa_main.c
+++ b/apps/uefi/sbsa_main.c
@@ -162,6 +162,7 @@ execute_tests()
         goto exit_acs;
     }
 
+    val_print(INFO, "\n\n SystemReady band ACS v3.1.1", 0);
     val_print(ERROR, "\n\n SBSA Architecture Compliance Suite\n");
     val_print(ERROR, "    Version %d.", SBSA_ACS_MAJOR_VER);
     val_print(ERROR, "%d.", SBSA_ACS_MINOR_VER);
