From 4c52e8d9d98adc88edfc382196b247820c387175 Mon Sep 17 00:00:00 2001
From: gurrev01 <gururaj.revankar@arm.com>
Date: Mon, 18 Dec 2023 19:53:31 +0530
Subject: [PATCH] enable psci checker

Signed-off-by: gurrev01 <gururaj.revankar@arm.com>
---
 drivers/firmware/psci/psci_checker.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/firmware/psci/psci_checker.c b/drivers/firmware/psci/psci_checker.c
index 116eb465c..0a80e8205 100644
--- a/drivers/firmware/psci/psci_checker.c
+++ b/drivers/firmware/psci/psci_checker.c
@@ -24,6 +24,7 @@
 #include <asm/cpuidle.h>
 
 #include <uapi/linux/psci.h>
+#include <linux/string.h>
 
 #define NUM_SUSPEND_CYCLE (10)
 
@@ -438,6 +439,15 @@ static int suspend_tests(void)
 static int __init psci_checker(void)
 {
 	int ret;
+    char *cmdline = boot_command_line; // Get kernel command line
+
+    // Check if "psci_checker" parameter is present in the kernel command line
+    if (strstr(cmdline, "psci_checker")) {
+        pr_info("PSCI checker enabled based on the kernel command line.\n");
+    } else {
+        pr_info("PSCI checker is disabled.\n");
+        return 0;
+    }
 
 	/*
 	 * Since we're in an initcall, we assume that all the CPUs that all
-- 
2.25.1

