diff --git a/uefi-sct/SctPkg/TestInfrastructure/SCT/Drivers/StandardTest/StandardTest.c b/uefi-sct/SctPkg/TestInfrastructure/SCT/Drivers/StandardTest/StandardTest.c
index 94cae289..5ecc0946 100644
--- a/uefi-sct/SctPkg/TestInfrastructure/SCT/Drivers/StandardTest/StandardTest.c
+++ b/uefi-sct/SctPkg/TestInfrastructure/SCT/Drivers/StandardTest/StandardTest.c
@@ -30,6 +30,7 @@ Abstract:
 #include "StandardTest.h"
 #include <Library/EntsLib.h>
 
+#define ACS_VERSION "SystemReady devicetree ACS v3.1.0\nBBR ACS 2.2.1 (EBBR)"
 static EFI_TIME Epoch = { .Year = 1970, .Month = 1, .Day = 1 };
 
 //
