diff --git a/uefi-sct/SctPkg/TestInfrastructure/SCT/Drivers/StandardTest/StandardTest.c b/uefi-sct/SctPkg/TestInfrastructure/SCT/Drivers/StandardTest/StandardTest.c
index 94cae289..21eec3b7 100644
--- a/uefi-sct/SctPkg/TestInfrastructure/SCT/Drivers/StandardTest/StandardTest.c
+++ b/uefi-sct/SctPkg/TestInfrastructure/SCT/Drivers/StandardTest/StandardTest.c
@@ -29,6 +29,9 @@ Abstract:
 #include "EfiTest.h"
 #include "StandardTest.h"
 #include <Library/EntsLib.h>
+
+#define ACS_VERSION "SystemReady IR ACS v2.0.0 Beta-1\nBBR ACS 1.0.2 (EBBR)"
+
 
 static EFI_TIME Epoch = { .Year = 1970, .Month = 1, .Day = 1 };
 
